compiler/il/symbol/OMRResolvedMethodSymbol.cpp

```diff
@@ -1419,7 +1419,69 @@ OMR::ResolvedMethodSymbol::getNumberOfBackEdges()
    return numBackEdges;
    }
 
+/*
+ * Returns the bytecode info for an OSR point, taking into account the
+ * possibility that the treetop for a call has different BCI to the OSR
+ * point.
+ */
+TR_ByteCodeInfo
+OMR::ResolvedMethodSymbol::getOSRByteCodeInfo(TR::Node *node)
+   {
+   TR_ByteCodeInfo nodeBCI = node->getByteCodeInfo();
+   if (node->getNumChildren() > 0 && (node->getOpCodeValue() == TR::treetop || node->getOpCode().isCheck()))
+      nodeBCI = node->getFirstChild()->getByteCodeInfo();
+   return nodeBCI;
+   }
+
+/*
+ * Checks the provided node is pending push store or load expected to be
+ * around an OSR point.
+ */
+bool
+OMR::ResolvedMethodSymbol::isOSRRelatedNode(TR::Node *node)
+   {
+   return (node->getOpCode().isStoreDirect()
+         && node->getOpCode().hasSymbolReference()
+         && node->getSymbolReference()->getSymbol()->isPendingPush())
+      || (node->getOpCodeValue() == TR::treetop
```

Comment:
During ILGen, anchored loads are generated to avoid the side effects of storing the stack into pending push temps. These anchored loads can exist between the OSR point and the induction point, so it is necessary to skip them.